robot/Cumulus/resources/ContactPageObject.py

```diff
@@ -76,8 +77,24 @@ def validate_relation_status_message(self, contact1, contact2, relation):
         id,actualstatus = self.npsp.check_status(contact1)
         self.builtin.should_be_equal_as_strings(actualstatus,expectedstatus)
 
-    # def validate_field_value_under_section(self, section, fieldname, value):
-    #     section="text:"+section
-    #     self.selenium.scroll_element_into_view(section)
-    #     self.npsp.confirm_field_value(fieldname, "contains", value)
+    def verify_rollup_field_value(self,field_name,value,section=None):
+        """Verifies if the given rollup field contains given value 
+        if it doesn't, then recalculates rollup and performs refresh"""
+        try :
+            self.npsp.navigate_to_and_validate_field_value(field_name,"contains",value,section)
+            self.builtin.log("Found rollup value on the page on initial try")
+        except Exception :
+            self.npsp.click_more_actions_button()
+            self.selenium.click_link("Recalculate Rollups")
+            self.npsp.wait_for_locator("frame", "accessibility title")
+            self.npsp.choose_frame("accessibility title")
+            self.selenium.wait_until_page_contains("The rollups batch job is calculating and may take a few moments to complete. Go back to the Contact and refresh to see updated rollup values.",timeout=60)
+            self.npsp.click_button_with_value("Back to Contact")
+            self.pageobjects.current_page_should_be("Details","Contact")
+            try :
+                self.npsp.navigate_to_and_validate_field_value(field_name,"contains",value,section)
+                self.builtin.log("Found rollup value on the page after recalculate")
+            except Exception :
+                self.selenium.reload_page()
+                self.npsp.navigate_to_and_validate_field_value(field_name,"contains",value,section)
```

Comment:
Would this be absolutely found on the 3rd try ?? If not, you can use a _for_ logic here to repeatedly try a few times (maybe 5 is a good number ??) and then add an exception at the end.
There is a similar logic I wrote in EDA repo if you wish to take a look. You can refer it at EDA.py file (keyword: select_frame_with_value)

Reply:
I wasn't sure on how much would be a good amount, on checking with the QE they said this should be good enough. Will definitely check it out if I notice that this is not enough